From 4ec6412b958b4848acec33a9f37a1d0041effd65 Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Fri, 12 Apr 2019 15:27:14 +0530
Subject: [PATCH 0127/1921] octeontx2-pf: Set RVU PF/VF watchdog timeout

Due to PSE deadlock errata, XOFF on TL2 transmission
queues takes more time than default watchdog timeout.
Hence setting watchdog timeout value high.

Change-Id: I28bbb045397e15a9a89140fae2cbbfbea3e617b4
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h | 7 +++++++
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c     | 1 +
 2 files changed, 8 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h
index 0909feb168f2..08175acbf447 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h
@@ -388,6 +388,13 @@ otx2_mbox_up_handler_ ## _fn_name(struct otx2_nic *pfvf,		\
 MBOX_UP_CGX_MESSAGES
 #undef M
 
+/* Time to wait before watchdog kicks off.
+ * Due to PSE deadlock errata, XOFF on TL2 transmission
+ * queues takes more time than default watchdog timeout.
+ * Hence setting this value higher.
+ */
+#define OTX2_TX_TIMEOUT		(100000 * HZ)
+
 #define	RVU_PFVF_PF_SHIFT	10
 #define	RVU_PFVF_PF_MASK	0x3F
 #define	RVU_PFVF_FUNC_SHIFT	0
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 7ed3e80714d5..cf558016312e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1245,6 +1245,7 @@ static int otx2_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 			       NETIF_F_HW_VLAN_CTAG_RX;
 
 	netdev->gso_max_segs = OTX2_MAX_GSO_SEGS;
+	netdev->watchdog_timeo = OTX2_TX_TIMEOUT;
 
 	netdev->netdev_ops = &otx2_netdev_ops;
 
-- 
2.31.1

